pkg/api/handlers/types/types.go

```diff
@@ -91,6 +91,11 @@ type ResponseGitOps struct {
 type ResponseCluster struct {
 	ID   string `json:"id"`
 	Slug string `json:"slug"`
+	// RequiresUpgrade represents whether the embedded cluster config for the current app
+	// version is different from the currently deployed embedded cluster config
+	RequiresUpgrade bool `json:"requiresUpgrade"`
+	// State represents the current state of the most recently deployed embedded cluster config
+	State string `json:"state,omitempty"`
```

Comment:
is there a more appropriate place for these response fields? `cluster` seems like an OK spot, but i know it's not necessarily meant to be for the "embedded cluster" but rather the "kots downstream cluster"